// figma-styled-components/dist/Tabs/Tabs.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tabs_1 = require("@reach/tabs");
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const SectionTitle_1 = require("../SectionTitle");
exports.Tabs = styled_components_1.default(tabs_1.Tabs) ``;
exports.TabList = styled_components_1.default(tabs_1.TabList) `
  border-bottom: solid 1px rgba(0,0,0,0.1);
`;
const StyledTab = styled_components_1.default(tabs_1.Tab) `
  border: none;
  background: #fff;
  outline: none;
  padding: 0 8px;
  ${SectionTitle_1.SectionTitle} {
    opacity: 0.5;
  }

  &[aria-selected='true'] {
    ${SectionTitle_1.SectionTitle} {
      opacity: 1;
    }
  }

  &:hover {
    ${SectionTitle_1.SectionTitle} {
      opacity: 1;
    }
  }
`;
exports.Tab = (props) => {
    return (react_1.default.createElement(StyledTab, null,
        react_1.default.createElement(SectionTitle_1.SectionTitle, null, props.children)));
};
exports.TabPanels = styled_components_1.default(tabs_1.TabPanels) ``;
exports.TabPanel = styled_components_1.default(tabs_1.TabPanel) ``;
