// figma-styled-components/dist/Checkbox/Checkbox.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const Text_1 = require("../Text");
const CheckboxMark = styled_components_1.default.span `
  position: absolute;
  top: 10px;
  left: 10px;
  width: 12px;
  height: 12px;
  border: 1px solid #000000;
  border-radius: 2px;
  background-color: #ffffff;
  box-sizing: border-box;

  &:after {
    position: absolute;
    width: 10px;
    height: 10px;
    content: "";
    background-image: url("data:image/svg+xml;utf8,%3Csvg%20fill%3D%22none%22%20height%3D%227%22%20viewBox%3D%220%200%208%207%22%20width%3D%228%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20clip-rule%3D%22evenodd%22%20d%3D%22m1.17647%201.88236%201.88235%201.88236%203.76471-3.76472%201.17647%201.17648-4.94118%204.9412-3.05882-3.05884z%22%20fill%3D%22%23fff%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E");
    background-repeat: no-repeat;
    background-position: center center;
  }
`;
const CheckboxContainerFacotry = ({ onChange, checked, value, ...props }) => {
    const handleChange = (event) => {
        if (onChange) {
            onChange(event);
        }
    };
    return (React.createElement("div", Object.assign({}, props),
        React.createElement("input", { type: 'checkbox', onChange: handleChange, checked: checked, value: value }),
        React.createElement(CheckboxMark, null)));
};
const CheckboxContainer = styled_components_1.default(CheckboxContainerFacotry) `
  position: relative;
  width: 32px;
  height: 32px;

  input {
    position: absolute;
    width: 0;
    height: 0;
    opacity: 0;
  }

  input:checked + ${CheckboxMark} {
    border: 1px solid #18a0fb;
    background-color: #18a0fb;
  }

  input:checked ${CheckboxMark}:after {
    display: block;
  }
`;
const CheckboxFactory = ({ onChange, label, checked, value, ...props }) => {
    return (React.createElement("label", Object.assign({}, props),
        React.createElement(CheckboxContainer, { onChange: onChange, checked: checked, value: value }),
        React.createElement(Text_1.Text, null, label)));
};
exports.Checkbox = styled_components_1.default(CheckboxFactory) `
  display: flex;
  flex-direction: row;
  height: 28px;
  cursor: default;
  align-items: center;
`;
