// figma-styled-components/dist/Section/Section.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const SectionTitle_1 = require("../SectionTitle");
class SectionFactory extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            collapsed: this.props.collapsed || false
        };
    }
    render() {
        return (React.createElement(SectionWrapper, Object.assign({ collapsed: this.state.collapsed }, this.props),
            React.createElement(SectionHeader, null,
                React.createElement(SectionTitle_1.SectionTitle, null, this.props.title),
                this.props.actions &&
                    React.createElement(SectionActions, null, this.props.actions)),
            React.createElement(SectionContent, null, this.props.children)));
    }
}
const SectionContent = styled_components_1.default.div ``;
const SectionActions = styled_components_1.default.div ``;
const SectionHeader = styled_components_1.default.div `
  display: flex;
  justify-content: space-between;
`;
const SectionWrapper = styled_components_1.default.section `
  padding: 8px;

  ${SectionHeader} {
    opacity: ${(props) => props.collapsed ? 0.5 : 1};
  }

  ${SectionContent} {
    display: ${(props) => props.collapsed ? 'none' : 'block'};
  }

  &:hover {
    ${SectionHeader} {
      opacity: 1;
    }
  }
`;
exports.Section = styled_components_1.default(SectionFactory) `


  & + & {
    border-top: solid 1px rgba(0,0,0, 0.1);
  }
`;
