// figma-styled-components/dist/Input/Input.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = __importDefault(require("styled-components"));
// TOOD: Figuure out how to make icon prop accept an item from Icon Names or a user supplied svg
// import { IconNames } from '../Icon'
const React = __importStar(require("react"));
const InputFactory = ({ icon, ...props }) => {
    return (React.createElement(exports.InputWrapper, { icon: icon },
        icon && React.createElement(exports.InputIconWrapper, { className: 'icon-holder' }, icon),
        React.createElement("input", Object.assign({}, props))));
};
exports.InputIconWrapper = styled_components_1.default.div `
  width: 32px;
  display: flex;
  align-items: center;
  justify-content: center;
  color: rgba(0, 0, 0, 0.5);
`;
exports.InputWrapper = styled_components_1.default.div `
  box-sizing: border-box;
  display: flex;
  align-items: center;
  width: 100%;
  height: 32px;
  margin: 1px 0 1px 0;
  padding: ${(props) => (props.icon ? '8px 4px 8px 0' : '8px 4px 8px 8px')};
  color: rgba(0, 0, 0, 0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  outline: none;
  font-family: 'Inter', sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  letter-spacing: 0.005em;

  input {
    border: none;
    outline: none;
    width: 100%;
    height: 24px;
  }

  &:hover {
    color: rgba(0, 0, 0, 0.8);
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
  }

  &:active,
  &:focus-within {
    color: #000000;
    box-shadow: inset 0 0 0 2px #18a0fb;
  }

  &::selection {
    color: #000000;
    background-color: rgba(24, 145, 251, 0.3);
  }

  &::placeholder {
    color: rgba(0, 0, 0, 0.3);
  }

  &:disabled {
    color: rgba(0, 0, 0, 0.3);
  }
`;
exports.Input = styled_components_1.default(InputFactory) ``;
