// figma-styled-components/dist/OptionButton/OptionButton.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
exports.OptionButtonContent = styled_components_1.default.div `
  align-items: center;
  border-radius: 2px;
  background: ${(props) => props.toggled ? 'rgba(0, 0, 0, 0.1)' : 'transparent'};
  border: none;
  display:flex;
  height: 24px;
  justify-content: center;
  opacity: ${(props) => props.disabled ? '0.3' : '1'};
  outline: none;
  pointer-events: none;
  text-align: center;
  min-width: 24px;
  width: 24px;
`;
const OptionButtonWrapper = styled_components_1.default.button `
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 2px;
  display: flex;
  height: 32px;
  justify-content: center;
  width: 32px;
  outline: none;
  color: #000;

  &:hover {
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1)
  }

  &:active ${exports.OptionButtonContent} {
    background: ${(props) => props.disabled ? 'transparent' : 'rgba(0, 0, 0, 0.1)'};
  }

  &:focus {
    box-shadow: inset 0 0 0 2px #18A0FB;
  }
`;
const OptionButtonFactory = ({ icon, toggled, onClick, disabled, ...props }) => {
    return (React.createElement(OptionButtonWrapper, Object.assign({ onClick: onClick, disabled: disabled || false }, props),
        React.createElement(exports.OptionButtonContent, { toggled: toggled || false, disabled: disabled || false }, icon)));
};
exports.OptionButton = styled_components_1.default(OptionButtonFactory) ``;
